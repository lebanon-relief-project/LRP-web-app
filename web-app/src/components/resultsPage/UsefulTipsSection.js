import React, { useState, useEffect } from "react";
import styled from "styled-components";
import colours from "../../styles/Colours";
import devices from "../../styles/Devices";
import PersonTakingCare from "../../assets/images/PersonTakingCare.svg";
import { getUsefulTips } from "../../services/results.service";

export const UsefulTipsSection = () => {
  const [usefulTips, setUsefulTips] = useState();

  const fetchUsefulTips = async () => {
    const response = await getUsefulTips();
    setUsefulTips(response);
  };

  useEffect(() => {
    fetchUsefulTips();
  }, []);

  return (
    <Wrapper>
      <StyledDiv>
        <div>
          <Title>A few more useful tips</Title>
          {usefulTips &&
            usefulTips.usefulTips.map((tip, i) => (
              <div key={i}>
                <h3>{tip.title}</h3>
                <p>{tip.body}</p>
              </div>
            ))}
        </div>
        <div>
          <img src={PersonTakingCare} alt={"Person Taking Care"} />
        </div>
      </StyledDiv>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  width: 60%;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: 0 16.7%;
  background: inherit;
  background: ${colours.lightGrey};

  @media (max-width: ${devices.ipadpro}) {
    margin: 25px 18px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
    max-width: 100%;
    padding: 0;
    margin: 0;
  }
`;

const StyledDiv = styled.div`
  display: flex;
  border-radius: 8px;
  padding: 5%;

  @media (max-width: ${devices.ipadpro}) {
    flex-direction: column;
    padding: 0;
    border-radius: 0;
    margin: 0 16px;
    width: 80%;
    margin: 0 auto;
  }

  @media (max-width: ${devices.mobile}) {
    width: 90%;
  }
`;

const Title = styled.h2`
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 38px;
`;

export default UsefulTipsSection;
