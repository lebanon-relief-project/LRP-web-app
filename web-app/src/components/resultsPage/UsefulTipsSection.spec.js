import React from "react";
import { render } from "@testing-library/react";
import UsefulTipsSection from "./UsefulTipsSection";
import "jest-styled-components";

describe("the UsefulTipsSection component", () => {
  let container;

  it("should match the snapshot", () => {
    ({ container } = render(<UsefulTipsSection />));

    expect(container).toMatchSnapshot();
  });
});
