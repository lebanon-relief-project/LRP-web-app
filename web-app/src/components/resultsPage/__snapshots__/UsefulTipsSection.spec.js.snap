// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UsefulTipsSection component should match the snapshot 1`] = `
.c0 {
  width: 60%;
  min-width: 960px;
  min-height: 730px;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: 0 16.7%;
  background: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fafafa;
  border-radius: 8px;
  padding: 5%;
}

.c2 {
  font-family: Playfair Display;
  font-style: normal;
  font-weight: bold;
  font-size: 30px;
  line-height: 38px;
}

@media (max-width:576px) {
  .c0 {
    margin: 25px 18px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
  }
}

@media (max-widthL 576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <h2
          class="c2"
        >
          A few more useful tips
        </h2>
      </div>
      <div>
        <img
          alt="Person Taking Care"
          src="PersonTakingCare.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
